src/transactionsFactories/transferTransactionsFactory.ts

```diff
@@ -152,6 +144,46 @@ export class TransferTransactionsFactory {
         }).build();
     }
 
+    createTransactionForTransfer(options: {
+        sender: IAddress;
+        receiver: IAddress;
+        nativeAmount?: bigint;
+        tokenTransfers?: TokenTransfer[];
+        data?: Uint8Array;
+    }): Transaction {
+        const nativeAmount = options.nativeAmount ?? 0n;
+        let tokenTransfers = options.tokenTransfers ? [...options.tokenTransfers] : [];
+        const numberOfTokens = tokenTransfers.length;
+
+        if (!nativeAmount && numberOfTokens === 0) {
+            throw new ErrBadUsage("No native token amount or token transfers provided");
```

Comment:
will this be thrown if I want to send a 'notarizing' transaction? since the native amount is 0 and there is no transferred token. I believe that this is a valid use case and we should allow it. So only throw error if nativeAmount is undefined and tokenTransfers is undefined

Reply:
ex: Alice wants to send Bob 0 egld but with data field 'Hello'. I think it will fail, as of now

Reply:
It will, but added support for it now.